guava/src/com/google/common/eventbus/AsyncEventBus.java

```diff
@@ -61,4 +61,112 @@ public AsyncEventBus(Executor executor, SubscriberExceptionHandler subscriberExc
   public AsyncEventBus(Executor executor) {
     super("default", executor, Dispatcher.legacyAsync(), LoggingHandler.INSTANCE);
   }
+
+  /**
+   * Returns a new builder. The generated builder is equivalent to the builder created by the {@link
+   * AsyncEventBus.Builder} constructor.
+   */
+  public static AsyncEventBus.Builder builder(final Executor executor) {
+    return new Builder(executor);
+  }
+
+  /**
+   * Creates a new AsyncEventBus that will use {@code executor} to dispatch events. Assigns {@code
+   * identifier} as the bus's name for logging purposes.
+   *
+   * @param identifier short name for the bus, for logging purposes.
+   * @param executor Executor to use to dispatch events. It is the caller's responsibility to shut
+   *     down the executor after the last event has been posted to this event bus.
+   * @param dispatcher dispatcher
+   * @param exceptionHandler Handler used to handle exceptions thrown from subscribers.
+   *     See {@link SubscriberExceptionHandler} for more information.
+   */
+  private AsyncEventBus(final String identifier,
```

Comment:
Should be package protected since this constructor is accessed from another class. Currently, JVM will generate an extra method to allow this call.